core/services/relay/evm/mercury/data_source.go

```diff
@@ -63,7 +70,35 @@ func (ds *datasource) Observe(ctx context.Context, repts ocrtypes.ReportTimestam
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			obs.MaxFinalizedBlockNumber.Val, obs.MaxFinalizedBlockNumber.Err = ds.fetcher.FetchInitialMaxFinalizedBlockNumber(ctx)
+			val, err := ds.fetcher.FetchInitialMaxFinalizedBlockNumber(ctx)
+			if err != nil {
+				obs.MaxFinalizedBlockNumber.Err = err
+				return
+			}
+			if val != nil {
+				obs.MaxFinalizedBlockNumber.Val = *val
+				return
+			}
+			if ds.initialBlockNumber == nil {
+				if obs.CurrentBlockNum.Err != nil {
+					obs.MaxFinalizedBlockNumber.Err = fmt.Errorf("FetchInitialMaxFinalizedBlockNumber returned empty LatestReport; this is a new feed. No initialBlockNumber was set, tried to use current block number to determine maxFinalizedBlockNumber but got error: %w", obs.CurrentBlockNum.Err)
+				} else {
+					// Subract 1 here because we will later add 1 to the
+					// maxFinalizedBlockNumber to get the first validFromBlockNum, which
+					// ought to be the same as current block num.
+					obs.MaxFinalizedBlockNumber.Val = obs.CurrentBlockNum.Val - 1
```

Comment:
Suggestion: How about using negative value (aka invalid) to denote `MaxFinalizedBlockNumber` for empty report and handle this case when calculating `validFromBlockNum`?

Reply:
`-1` is already used as a value to indicate that we want the feed to start from 0